benchmarks/overrides_benchmark/common.py

```diff
@@ -28,4 +28,4 @@ def __torch_function__(self, func, types, args=(), kwargs=None):
         if kwargs is None:
             kwargs = {}
 
-        return args[0] + args[1]
+        return super().__torch_function__(func, types, args, kwargs)
```

Comment:
This doesn't look semantics preserving...

Reply:
It would recurse infinitely without this change as `args[0].__add__` would come back into `__torch_function__`.

Reply:
Let's spell out:
1. Under which circumstance does the behavior change (e.g. `Tensor` subclasses outside of PyTorch that have a `__torch_function__` + concrete example).
2. What the failure mode is (infinite recursion).
3. How to deal with this (docs for 1.7.0; possibility of raising a clear error, etc.)

Reply:
So, the issue is specifically with `torch.Tensor` subclasses that already implement `__torch_function__`, and also call `torch.Tensor` methods. Concretely, consider the following subclass:

```python
class SubTensor(torch.Tensor):
    def __torch_function__(self, func, types, args=(), kwargs=None):
        if kwargs is None:
            kwargs = {}
        if func is torch.add:
            return (args[0] + args[1]).as_subclass(SubTensor)
```

Previously, this would work fine, as `operator.add` wouldn't recurse into `__torch_function__`. But now it does, and a call to `super().__torch_function__` would be needed to fix this issue.

Reply:
I think the way to deal with it is just documentation.

Reply:
Let's make sure to add a note to `Tensor.__torch_function__` then, as well as an explanation to https://pytorch.org/docs/stable/notes/extending.html#extending-torch. It looks like the examples in those docs don't suffer from this. I suspect the impact will be limited.

Reply:
Done.